staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset_test.go

```diff
@@ -659,59 +680,83 @@ func TestContextCancel(t *testing.T) {
 		t.Fatal(err)
 	}
 	qs := qsc.Complete(fq.DispatchingConfig{ConcurrencyLimit: 1})
-	counter.Add(1) // account for the goroutine running this test
+	counter.Add(1) // account for main activity of the goroutine running this test
 	ctx1 := context.Background()
 	b2i := map[bool]int{false: 0, true: 1}
-	var qnc [2][2]int32
-	req1, _ := qs.StartRequest(ctx1, &fcrequest.WorkEstimate{Seats: 1}, 1, "", "fs1", "test", "one", func(inQueue bool) { atomic.AddInt32(&qnc[0][b2i[inQueue]], 1) })
+	var qnc [2][2]int32 // counts of calls to the following queueNoteFns
+	queueNoteFn1 := func(inQueue bool) { atomic.AddInt32(&qnc[0][b2i[inQueue]], 1) }
+	queueNoteFn2 := func(inQueue bool) { atomic.AddInt32(&qnc[1][b2i[inQueue]], 1) }
+	expectQNCounts := func(fn int, expectF, expectT int32) {
+		if a := atomic.LoadInt32(&qnc[fn-1][0]); a != expectF {
+			t.Errorf("Got %d calls to queueNoteFn%d(false), expected %d", a, fn, expectF)
+		}
+		if a := atomic.LoadInt32(&qnc[fn-1][1]); a != expectT {
+			t.Errorf("Got %d calls to queueNoteFn%d(true), expected %d", a, fn, expectT)
+		}
+	}
+	req1, _ := qs.StartRequest(ctx1, &fcrequest.WorkEstimate{Seats: 1}, 1, "", "fs1", "test", "one", queueNoteFn1)
 	if req1 == nil {
 		t.Error("Request rejected")
 		return
 	}
-	if a := atomic.AddInt32(&qnc[0][0], 0); a != 1 {
-		t.Errorf("Got %d calls to queueNoteFn1(false), expected 1", a)
-	}
-	if a := atomic.AddInt32(&qnc[0][1], 0); a != 1 {
-		t.Errorf("Got %d calls to queueNoteFn1(true), expected 1", a)
+	expectQNCounts(1, 1, 1)
+	fatalErrs := []string{}
+	var errsLock sync.Mutex
+	expectQNCount := func(fn int, inQueue bool, expect int32) {
+		if a := atomic.LoadInt32(&qnc[fn-1][b2i[inQueue]]); a != expect {
+			errsLock.Lock()
+			defer errsLock.Unlock()
+			fatalErrs = append(fatalErrs, fmt.Sprintf("Got %d calls to queueNoteFn%d(%v), expected %d", a, fn, inQueue, expect))
```

Comment:
IIUC, you need to lock here to avoid concurrent t.Errorf() calls.

Given you need it, let's remove the expecteQNCounts() function above and use just this one.

Reply:
done